frontend/ui/widget/sortwidget.lua

```diff
@@ -0,0 +1,499 @@
+        button_vertical_line,
+        self.footer_right,
+        button_vertical_line,
+        self.footer_last_down,
+        button_vertical_line,
+        self.footer_ok,
+    }
+    local bottom_line = LineWidget:new{
+        dimen = Geom:new{ w = self.item_width, h = Size.line.thick },
+        background = Blitbuffer.COLOR_DARK_GRAY,
+        style = "solid",
+    }
+    local vertical_footer = VerticalGroup:new{
+        bottom_line,
+        self.page_info
+    }
+    local footer = BottomContainer:new{
+        dimen = self.dimen:copy(),
+        vertical_footer,
+    }
+    -- setup title bar
+    self.title_bar = SortTitleWidget:new{
+        title = self.title,
+        width = self.item_width,
+        height = self.item_height,
+        use_top_page_count = self.use_top_page_count,
+        sort_page = self,
+    }
+    -- setup main content
+    self.item_margin = self.item_height / 8
+    local line_height = self.item_height + self.item_margin
+    local content_height = self.dimen.h - self.title_bar:getSize().h - vertical_footer:getSize().h - padding
+    self.items_per_page = math.floor(content_height / line_height)
+    self.pages = math.ceil(#self.item_table / self.items_per_page)
+    self.main_content = VerticalGroup:new{}
+
+    self:_populateItems()
+
+    local frame_content = FrameContainer:new{
+        height = self.dimen.h,
+        padding = padding,
+        bordersize = 0,
+        background = Blitbuffer.COLOR_WHITE,
+        VerticalGroup:new{
+            align = "left",
+            self.title_bar,
+            self.main_content,
+        },
+    }
+    local content = OverlapGroup:new{
+        dimen = self.dimen:copy(),
+        frame_content,
+        footer,
+    }
+    -- assemble page
+    self[1] = FrameContainer:new{
+        height = self.dimen.h,
+        padding = 0,
+        bordersize = 0,
+        background = Blitbuffer.COLOR_WHITE,
+        content
+    }
+end
+
+function SortWidget:nextPage()
+    local new_page = math.min(self.show_page+1, self.pages)
+    if new_page > self.show_page then
+        self.show_page = new_page
+        if self.marked > 0 then
+            self:swipeItem(self.items_per_page * (self.show_page - 1) + 1 - self.marked)
+        end
+        self:_populateItems()
+    end
+end
+
+function SortWidget:prevPage()
+    local new_page = math.max(self.show_page-1, 1)
+    if new_page < self.show_page then
+        self.show_page = new_page
+        if self.marked > 0 then
+            self:swipeItem(self.items_per_page * (self.show_page - 1) + 1 - self.marked)
+        end
+        self:_populateItems()
+    end
+end
+
+function SortWidget:goToPage(page)
+    self.show_page = page
+    self:_populateItems()
+end
+
+function SortWidget:swipeItem(diff)
+    local swipe_to = self.marked + diff
+    if swipe_to > 0 and swipe_to <= #self.item_table then
+        self.show_page = math.ceil(swipe_to/self.items_per_page)
+        self:swipeItems(self.marked, swipe_to)
```

Comment:
I was thinking you were supporting the swipe gesture when reading this :) and was about to suggest it's better be named a pan :) But no.
Also, got confused by `self:swipeItem()` calling `self:swipeItems()` recursively (I missed the additional **s**) :)

Better this one be named `SortWidget:moveItem(diff)` and the other one below: `SortWidget:moveItemTo(pos1, pos2)`, if they are actually just moving items :)

OR, if I get it better after a re-read :) the one below should be named `SortWidget:swapItems(pos1, pos2)` which I think is what you meant :)